pkgs/shells/zsh/zsh-history/default.nix

```
@@ -0,0 +1,32 @@
+{ stdenv, lib, fetchFromGitHub, buildGoPackage }:
+
+with lib;
+
+buildGoPackage rec {
+  pname = "zsh-history";
+  version = "2019-10-27";
+
+  src = fetchFromGitHub {
+    owner = "b4b4r07";
+    repo = "history";
+    rev = "527e6f51873992fbf9c1aad70aa3009a0027a8de";
+    sha256 = "12dc380zfg3b9k7rcsyzi9dxqh28c4957b3fsx1nxvqvdm3ralm2";
+  };
+
+  goDeps = ./deps.nix;
+  goPackagePath = "github.com/b4b4r07/history";
+
+  postInstall = ''
+    install -d $out/share
+    cp -r $NIX_BUILD_TOP/go/src/${goPackagePath}/misc/* $out/share
+    cp -r $out/share/zsh/completions $out/share/zsh/site-functions
+  '';
+
+  meta = {
+    description = "A CLI to provide enhanced history for your shell";
```

Which shells does it work with?
Just fish and zsh or also others?

---

It may work with other, but afaik zsh is the only one that is actively supported.
There is an outstanding issue for fish, for example, that prevents it from functioning properly (https://github.com/b4b4r07/history/pull/3).
As I can only attest to the zsh integration, I am not comfortable suggesting it for anything else.

---

@kampka, this is under `shells/zsh` so I'd assume immediately that zsh is only supported. If there's outstanding issues with other shells they're probably not supported, or supported well.

```suggestion
    description = "A CLI to provide enhanced history for your ZSH shell";
```

---

@worldofpeace yes indeed, the author is quite engaged in the ZSH ecosystem, any other shell support is mostly community driven contributions here and there, hence I'm uncomfortable suggesting using it in fish or similar shell atm.